src/azure-cli/azure/cli/command_modules/role/_validators.py

```diff
@@ -91,6 +91,18 @@ def process_assignment_namespace(cmd, namespace):  # pylint: disable=unused-argu
     if namespace.scope and resource_group and getattr(resource_group, 'is_default', None):
         namespace.resource_group_name = None  # drop configured defaults
 
+    if not namespace.scope and not namespace.resource_group_name:
+        raise CLIError('usage error: please specify at least one of "--scope" and "--resource-group".')
+
+
+def process_list_assignment_namespace(cmd, namespace):  # pylint: disable=unused-argument
+    resource_group = namespace.resource_group_name
+    if namespace.scope and resource_group and getattr(resource_group, 'is_default', None):
+        namespace.resource_group_name = None  # drop configured defaults
+
+    if not namespace.show_all and not namespace.scope and not namespace.resource_group_name:
+        raise CLIError('usage error: please specify at least one of "--all", "--scope" and "--resource-group".')
```

Comment:
Why the behavior of `az role assignment list` is also changed? There doesn't seem to be any security concern in this command.

Also, it is also a breaking change to force one of `--all`, `--scope` and `-resource-group`, which needs to go to the history.